src/Components/LogupLayout/logupForm.js

```diff
@@ -0,0 +1,83 @@
+import React, { useRef, useState } from 'react';
+
+const LogupForm = () => {
+  const [email, setEmail] = useState('');
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+  const [fullName, setFullName] = useState('');
+  const [users, setUsers] = useState(
+    localStorage.registeredUsers ? JSON.parse(localStorage.getItem('registeredUsers')) : []
+  );
+
+  const errMessageRef = useRef(null);
+
+  localStorage.setItem('registeredUsers', JSON.stringify(users));
+
+  const usernameRegexp = /^[0-9A-Za-z]{6,16}$/;
```

Comment:
move this outside of function component, you're creating it after every re-render